internal/viridian/cluster.go

```diff
@@ -0,0 +1,120 @@
+package viridian
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/rand"
+	"os"
+	"path"
+	"time"
+)
+
+type createClusterRequest struct {
+	KubernetesClusterID int         `json:"kubernetesClusterId"`
+	Name                string      `json:"name"`
+	ClusterTypeID       ClusterType `json:"clusterTypeId"`
+	PlanName            ClusterPlan `json:"planName"`
+}
+
+type createClusterResponse Cluster
+
+func (a API) CreateCluster(ctx context.Context, name string, plan ClusterPlan, k8sClusterID int, isDev bool) (Cluster, error) {
+	if name == "" {
+		name = clusterName()
+		if name == "" {
+			return Cluster{}, errors.New("could not generate cluster name")
+		}
```

Comment:
I think `clusterName` should always return a name, so this check can be removed.